pingrid.py

```diff
@@ -27,6 +27,17 @@
 RAIN_PNE_COLORMAP = "[0xa020f0 [0x0000ff 35] [0x40e0d0 36] [0x32cd32 36] 0xffe465 [0xffe465 35] 0xffff00 [0xffa500 35] [0xa52a2a 36] [0x000000 36]]"
 CORRELATION_COLORMAP = "[0x000000 0x000080 [0x0000ff 25] [0x00bfff 26] [0x7fffd4 39] [0x98fb98 26] 0xffe465 [0xffe465 25] 0xffff00 [0xff8c00 38] [0xff0000 39] [0x800000 39] 0xa52a2a]"
 
+
+def sel_snap(spatial_array, lat, lng, dim_y="Y", dim_x="X"):
+    """"Selects the spatial_array's closest spatial grid center to the lng/lat coordinate.
+    Raises an excpetion if lng/lat is outside spatial_array domain.
+    Expects spatial grids to be square.
```

Comment:
True. I hadn't thought about that. We don't need to make that assumption: we could first check if the point is inside the bounds of the dataset, and if so, use `.sel(method=nearest)` with no tolerance limit.

That version would also be more correct than this one even in the case of square grid cells. The area defined by the cell center and the tolerance is a circle, not a square; for a pixel on the edge of the dataset, there are points that are inside the tolerance circle but outside the dataset.

Reply:
Oh, and once we've generalized it to non-square grid cells, we don't have to limit it to spatial dimensions anymore. We could just take arbitrary values for arbitrary coordinates. (I'm guessing this generalization might be hard for you to code. You can leave it for another time.)

Reply:
That all makes sense. Testing whether the point is inside the bounds of the dataset may (or not) be tricky for boundless periodic domains, e.g. global longitude. xarray is not dealing with that, is it? It may have longitude going from -180 to 180 and other from 0 to 360, correct? Then -1 will be in bounds of the first case and out bounds of the 2nd case. It's probably fine to work within such a framework until we come up with periodicity considerations. Or maybe xarray takes care of that?

Reply:
How does that look now?

Obstacles to more generalization:
1/ dims must be evenly spaced for (half-)resolution to exist
2/ dims expected to be in increasing order -- that should be generalizable without much trouble
3/ generalize to apply to n dimensions? how would that work? the input is then a dict rather than a list of values and dims?